axonserver/src/test/java/io/axoniq/axonserver/component/version/BackwardsCompatibleVersionTest.java

```diff
@@ -0,0 +1,138 @@
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version5_1_3);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version3_1_3);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version4_1);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version4_2);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version4_0);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version5_1);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version3_1);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version4);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version3);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1.greaterOrEqualThan(version5);
+        assertFalse(greaterOrEqual);
+    }
+
+    @Test
+    public void testGreaterOrEqualThanWithPatch() {
+        boolean greaterOrEqual = version4_1_3.greaterOrEqualThan(version4_1_3);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1_3.greaterOrEqualThan(version4_1_2);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1_3.greaterOrEqualThan(version4_1_4);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1_3.greaterOrEqualThan(version5_1_3);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1_3.greaterOrEqualThan(version3_1_3);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1_3.greaterOrEqualThan(version4_1);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1_3.greaterOrEqualThan(version4_2);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1_3.greaterOrEqualThan(version4_0);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1_3.greaterOrEqualThan(version5_1);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1_3.greaterOrEqualThan(version3_1);
+        assertFalse(greaterOrEqual);
+
+        greaterOrEqual = version4_1_3.greaterOrEqualThan(version4);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1_3.greaterOrEqualThan(version3);
+        assertTrue(greaterOrEqual);
+
+        greaterOrEqual = version4_1_3.greaterOrEqualThan(version5);
+        assertFalse(greaterOrEqual);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testTooManyVersionNumbersInVersionClass() {
+        version4_1_3.greaterOrEqualThan(new BackwardsCompatibleVersion("4.1.3.5"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testTooFewVersionNumbersInVersionClass() {
```

Comment:
public void testTooManyVersionNumbersInVersionClass()
and
    public void testTooFewVersionNumbersInVersionClass() 
looks the same